python/src/aac/lang/definitions/definition.py

```diff
@@ -82,12 +90,14 @@ def get_top_level_fields(self) -> dict[str, dict]:
     def get_description(self) -> Optional[str]:
         """Return the description for the current definition, or None if it isn't defined."""
         fields = self.get_top_level_fields()
-        return fields.get(DEFINITION_FIELD_DESCRIPTION)
+        description = fields.get(DEFINITION_FIELD_DESCRIPTION)
+        return str(description) if description else None
 
     def get_type(self) -> Optional[str]:
         """Return the string for the extension type field, or None if the field isn't defined."""
         fields = self.get_top_level_fields()
-        return fields.get(DEFINITION_FIELD_TYPE)
+        type_field = fields.get(DEFINITION_FIELD_TYPE)
+        return str(type_field) if type_field else None
```

Comment:
This is to enforce that the returned type is actually [str, None] since a dict can return a non-string value and downstream functions may assume non-None is a str and therefore capable of performing string methods which obviously throw errors if the object returned from the dict isn't a string.